File: 2120-execution-of-all-suffix-instructions-staying-in-a-grid/2120-execution-of-all-suffix-instructions-staying-in-a-grid.py
class Solution:
    def executeInstructions(self, n: int, startPos: List[int], s: str) -> List[int]:
        mp, res = {"L": [0, -1], "R": [0, 1], "U": [-1, 0], "D": [1, 0]}, []
        for i in range(0, len(s)):
            cur_r, cur_c = startPos
            moves = 0
            for j in range(i, len(s)):
                r, c = mp[s[j]]
                cur_r += r
                cur_c += c
                if cur_r < 0 or cur_r >= n or cur_c < 0 or cur_c >= n:
                    break
                moves += 1
            res.append(moves)
        return res
        

File: 2120-execution-of-all-suffix-instructions-staying-in-a-grid/2120-execution-of-all-suffix-instructions-staying-in-a-grid.js
/**
 * @param {number} n
 * @param {number[]} startPos
 * @param {string} s
 * @return {number[]}
 */
var executeInstructions = function(n, startPos, s) {
    const map = {"R": [0, 1], "L": [0, -1], "U": [-1, 0], "D": [1, 0]}
    const res = []
    for (let i=0; i < s.length; i++) {
        let [curR, curC] = startPos, moves = 0
        for (let j=i; j < s.length; j++) {
            const [r, c] = map[s[j]]
            curR += r
            curC += c
            if (curR < 0 || curR >= n || curC < 0 || curC >= n) break
            moves++
        }
        res.push(moves)
    }
    return res
};